engine/crates/engine-v2/src/response/error.rs

```diff
@@ -1,44 +1,68 @@
-use std::collections::BTreeMap;
+use std::borrow::Cow;
 
-use engine::ErrorCode;
+pub(crate) use engine::ErrorCode;
+
+use crate::operation::Location;
 
 use super::ResponsePath;
 
-#[derive(Debug, Clone, Default)]
+#[derive(Debug, Clone)]
 pub(crate) struct GraphqlError {
-    pub message: String,
-    pub locations: Vec<crate::operation::Location>,
+    pub message: Cow<'static, str>,
+    pub code: ErrorCode,
+    pub locations: Vec<Location>,
     pub path: Option<ResponsePath>,
-    // ensures consistent ordering for tests
-    pub extensions: BTreeMap<String, serde_json::Value>,
+    // Serialized as a map, but kept as a Vec for efficiency.
+    // "code" must not be present
+    pub extensions: Vec<(Cow<'static, str>, serde_json::Value)>,
```

Comment:
This is kind of breaking, because the btree is ordered by the key and this is not. It's probably fine, just hilighting this here.

Reply:
it's serialized as a map, no ordering guarantees in GraphQL, but yeah fair point